sw/legacy/demo/hello_world/main.c

```diff
@@ -9,28 +9,63 @@
 #include "pwm.h"
 #include "timer.h"
 
+/**
+ * Is the code running in simulation (as opposed to on FPGA)?
+ */
+#define SIMULATION 0
+
 #define USE_GPIO_SHIFT_REG 0
 
+static const bool dual_uart = true;
+static uart_t uart0, uart1;
+
 void test_uart_irq_handler(void) __attribute__((interrupt));
 
 void test_uart_irq_handler(void) {
   int uart_in_char;
 
   while ((uart_in_char = uart_in(DEFAULT_UART)) != -1) {
-    uart_out(DEFAULT_UART, uart_in_char);
-    uart_out(DEFAULT_UART, '\r');
-    uart_out(DEFAULT_UART, '\n');
+    uart_out(uart0, uart_in_char);
+    uart_out(uart0, '\r');
+    uart_out(uart0, '\n');
   }
 }
 
+static inline void write_both(char ch0, char ch1) {
+  uart_out(uart0, ch0);
+  uart_out(uart1, ch1);
+}
+
 int main(void) {
 //  install_exception_handler(UART_IRQ_NUM, &test_uart_irq_handler);
 //  uart_enable_rx_int();
-  uart_init(DEFAULT_UART);
+  if (dual_uart) {
+    const char *signon = "hello_world demo application; UART ";
+    uart0 = UART_FROM_BASE_ADDR(UART0_BASE);
```

Comment:
Any reason this is a static variable rather than a macro definition in `sonata_system.h`?